pkg/util/sets/ipnet.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"net"
+	"strings"
+)
+
+type IPNet map[string]*net.IPNet
```

Comment:
OK I'll move it.  Shadowing net is naughty (util/net), but it looks like the best place.  But if we are shadowing anyway, how about putting it under net (util/net/sets), so it will be netsets.IPNet?

Pretty sure it can't be used as a map key, because you can't use anything with []byte as a map key, and IP is a []byte - the workaround is to use a string representation.

With pointers retrieval doesn't work: http://play.golang.org/p/Mc1BO9T5GH

Trying to use `map[net.IPNet]struct{}` gives `invalid map key type net.IPNet`: http://play.golang.org/p/qrw1UvQ6tH

Another option would be to create a string-backed IPNet e.g. http://play.golang.org/p/Mbp-AH-Isb  But really this should just be fixed at the language level...